src/js/components/Diagram/Diagram.js

```diff
@@ -66,198 +63,200 @@ const findTarget = (target) => {
   return target;
 };
 
-const Diagram = forwardRef(({ connections, ...rest }, ref) => {
-  const theme = useContext(ThemeContext) || defaultProps.theme;
-  const [dimensions, setDimensions] = useState({ width: 0, height: 0 });
-  const [connectionPoints, setConnectionPoints] = useState();
-  const svgRef = useForwardedRef(ref);
+const defaultConnectionsValue = [];
```

Comment:
why was this added?

Reply:
It's added in order to keep parity with `defaultProps` approach and avoid potentially redundant rerenders or even rerender loops.
Let's consider a prop with value of object type. Assume it has no value passed and falls back to value from `defaultProps`. In this case, the value would be equal between renders (`props.prop === defaultProps.prop`), since `defaultProps` is static.
If we replace `defaultProps` with default function parameters, and the default value of mentioned prop is defined as object literal, this means that on each render its values would be different. On each render component function is called again, so default parameter value is re-evaluated. If component state depends on such prop (e.g., via `useEffect` or `useMemo`), this would result in infinite rerenders loop.
That's why I've initialized such props with stable links. Actually this approach was already used in some other places of grommet.

Reply:
Thank you ☝️